source/server/server.cc

```diff
@@ -158,6 +165,10 @@ void InstanceImpl::flushStats() {
   });
 }
 
+void InstanceImpl::RegisterMutexContentionStats() {
+  absl::RegisterMutexTracer(&Envoy::MutexTracer::ContentionHook);
```

Comment:
This absl:: call should IMO be done from MutexTracer's ctor which should only be called once per process. Here we should have:

`mutex_tracer_ = MutexTracer::getOrCreateTracer();`

which should return the static pointer after creating it if needed. The admin endpoint can then get the tracer from the server when /contention is queried.